server/commands/shareLoader.ts

```diff
@@ -0,0 +1,231 @@
+import { Op, WhereOptions } from "sequelize";
+import isUUID from "validator/lib/isUUID";
+import { NavigationNode } from "@shared/types";
+import { UrlHelper } from "@shared/utils/UrlHelper";
+import { AuthorizationError, NotFoundError } from "@server/errors";
+import { Collection, Document, Share, User } from "@server/models";
+import { authorize, can } from "@server/policies";
+
+type LoadPublicShareProps = {
+  id: string;
+  collectionId?: string;
+  documentId?: string;
+  teamId?: string;
+};
+
+export async function loadPublicShare({
+  id,
+  collectionId,
+  documentId,
+  teamId,
+}: LoadPublicShareProps) {
+  const urlId =
+    !isUUID(id) && UrlHelper.SHARE_URL_SLUG_REGEX.test(id) ? id : undefined;
+
+  if (urlId && !teamId) {
+    throw new Error("teamId required for fetching share using urlId");
+  }
+
+  const where: WhereOptions<Share> = {
+    revokedAt: {
+      [Op.is]: null,
+    },
+    published: true,
+  };
+
+  if (urlId) {
+    where.urlId = id;
+    where.teamId = teamId;
+  } else {
+    where.id = id;
+  }
+
+  const share = await Share.findOne({
+    where,
+    include: [
+      {
+        model: Document.scope("withDrafts"),
```

Comment:
I don't think drafts should be included – what is shared should represent the tree in Outline?